src/screens/credits.rs

```diff
@@ -5,20 +5,23 @@ use bevy::prelude::*;
 use super::Screen;
 use crate::{
     game::{assets::SoundtrackKey, audio::soundtrack::PlaySoundtrack},
+    spawn::prelude::*,
     ui::prelude::*,
 };
 
 pub(super) fn plugin(app: &mut App) {
-    app.add_systems(OnEnter(Screen::Credits), show_credits_screen);
-    app.add_systems(OnExit(Screen::Credits), disable_soundtrack);
+    app.add_systems(
+        OnEnter(Screen::Credits),
+        (credits_screen.spawn(), start_credits_soundtrack),
```

Comment:
Calling `spawn()` here and in similar places *looks* like a bug. I know it isn't, but it very very much looks like we accidentally just called the spawn method.

Reply:
Well, originally I was going to leave the extension trait that adds this out of the PR, in which case you would have to write a little `fn spawn_credits_screen` system every time. Which isn't too bad. I'm fine with removing `SpawnSystemExt`.

Reply:
I think that would help reduce the complexity a bit.

Reply:
Removed `SpawnSystemExt`.